workflows/crisprseq_targeted.nf

```diff
@@ -0,0 +1,774 @@
+            .join(SEQTK_SEQ_MASK.out.fastx)
+            .join(CUTADAPT.out.log)
+            .set { ch_merging_summary_data }
+    } else {
+        ch_cat_fastq.paired
+            .mix(ch_cat_fastq.single)
+            .join(PEAR.out.assembled, remainder: true)
+            .join(SEQTK_SEQ_MASK.out.fastx)
+            .combine(Channel.value("null"))
+            .map { meta, reads, assembled, masked, trimmed ->
+                if (assembled == null) {
+                    assembled = file('null_a')
+                }
+                if (trimmed == "null") {
+                    trimmed = file('null_t')
+                }
+                return [ meta, reads, assembled, masked, trimmed ]
+            }
+            .set { ch_merging_summary_data }
+    }
+
+    //
+    // MODULE: Summary of merged reads
+    //
+    MERGING_SUMMARY {
+        ch_merging_summary_data
+    }
+
+
+    if (params.umi_clustering) {
+        //
+        // MODULE: Extract UMI sequences
+        //
+        EXTRACT_UMIS (
+            SEQTK_SEQ_MASK.out.fastx
+        )
+
+
+        //
+        // MODULE: Cluster UMIs
+        //
+        VSEARCH_CLUSTER (
+            EXTRACT_UMIS.out.fasta
+        )
+
+        //  Obtain a file with UBS (UBI bin size) and UMI ID
+        VSEARCH_CLUSTER.out.clusters
+        .transpose()
+        .collectFile( storeDir:params.outdir ) {
+            it ->
+                [ "${it[0].id}_ubs.txt", "${it[1].countFasta()}\t${it[1].baseName}\n" ]
+        }
+
+        // Branch the clusters into the ones containing only one sequence and the ones containing more than one sequences
+        VSEARCH_CLUSTER.out.clusters
+        .transpose()
+        .branch{
+            meta, cluster ->
+                single: cluster.countFasta() >= params.umi_bin_size && cluster.countFasta() == 1
+                    return [meta, cluster]
+                cluster: cluster.countFasta() >= params.umi_bin_size && cluster.countFasta() > 1
+                    return [meta, cluster]
+        }
+        .set{ ch_umi_bysize }
+
+        // Get the correspondent fasta sequencences from single clusters
+        ch_umi_bysize.single
+        .tap{ meta_channel }
+        .map{ meta, cluster ->
+            fasta_line = umi_to_sequence(cluster)
+            [meta, cluster.baseName, fasta_line]
+        }
+        .collectFile() { meta, name, fasta ->
+            [ "${name}_consensus.fasta", fasta ]
+        }
+        .map{ new_file ->
+            [new_file.baseName, new_file]
+        }
+        .join(meta_channel
+            .map { meta, original_file ->
+                ["${original_file.baseName}_consensus", meta]
+            })
+        .map{ file_name, new_file, meta ->
+            [meta, new_file]
+        }
+        .set{ ch_single_clusters_consensus }
+
+        //
+        // MODULE: Obtain most abundant UMI in cluster
+        //
+        VSEARCH_SORT(
+            ch_umi_bysize.cluster,
+            Channel.value("--sortbysize")
```

Comment:
Never seen this before 👀  I'm just curious, could this also be done with `params.sortbysize` with an initialized Channel at the beginning of the workflow? Like this [here](https://github.com/nf-core/sarek/blob/ed1cc8499366dcefea216fe37e36c6189537d57b/workflows/sarek.nf#L143)?

Reply:
Definitely :) But that is pipeline specific, the module must run with this parameter, so the users shouldn't be able to change this parameter

Reply:
Ah I see, could this also to into the `modules.config` then?

Reply:
After in person discussions with @SusiJo, we decided that we will do two separate modules in the future: `vsearch/sortbysize` & `vsearch/sortbyabundance`

Reply:
Will it be good to set hidden parameter or use module config for this?

Reply:
What do you mean by hidden parameter? I would prefer to not give the possibility to a user modifying this. It is also difficult to give it through modules.config as it is a channel.
The best solution would be to split the module in two different modules (one for each of the possibilities for this argument). It's a mandatory and positional argument of the tool, so it must be provided that way, but I didn't consider the option of having two subtools when I created the module. This solution would imply updating the module in nf-core/modules though, so I would prefer to do it for the next version of the pipeline :)